compiler/rustc_data_structures/src/fingerprint.rs

```diff
@@ -93,7 +104,7 @@ impl FingerprintHasher for crate::unhash::Unhasher {
     #[inline]
     fn write_fingerprint(&mut self, fingerprint: &Fingerprint) {
         // `Unhasher` only wants a single `u64`
-        self.write_u64(fingerprint.0);
+        self.write_u64(fingerprint.0.wrapping_add(fingerprint.1));
```

Comment:
`to_smaller_hash()` is a bit more expensive because it multiplies one of the values. This multiplication leads to different values depending on the order of the two components (something that we want in a stable hashing context). `Unhasher` on the other hand is just used for hash table, so it's fine if `Fingerprint(x,y)` and `Fingerprint(y,x)` result in the same hash value. 

Let me add a comment about that.